compiler/haskell-ide-core/src/Development/IDE/Types/Diagnostics.hs

```diff
@@ -3,137 +3,230 @@
 
```

Comment:
Should some of the serious/misclassified error stuff in Shake.hs move in here as a nice abstract API? (Perhaps during a follow-up)

Reply:
Sure, it makes a lot more sense to do that when we move to the DiagnosticsStore